spring-boot/proxy/src/main/java/eu/coatrack/proxy/security/ApiKeyAuthTokenVerifier.java

```diff
@@ -87,17 +65,18 @@ public Authentication authenticate(Authentication authentication) throws Authent
             if (apiKey.equals(ApiKey.API_KEY_FOR_YGG_ADMIN_TO_ACCESS_PROXIES)) {
 
                 Set<SimpleGrantedAuthority> authoritiesGrantedToYggAdmin = new HashSet<>();
-                authoritiesGrantedToYggAdmin.add(new SimpleGrantedAuthority(ServiceApiAccessRightsVoter.ACCESS_SERVICE_AUTHORITY_PREFIX + "refresh"));
+                authoritiesGrantedToYggAdmin.add(new SimpleGrantedAuthority(
+                        ServiceApiAccessRightsVoter.ACCESS_SERVICE_AUTHORITY_PREFIX + "refresh"));
 
                 ApiKeyAuthToken apiKeyAuthToken = new ApiKeyAuthToken(apiKey, authoritiesGrantedToYggAdmin);
                 apiKeyAuthToken.setAuthenticated(true);
                 return apiKeyAuthToken;
             }
 
             // regular api consumer's api key check
-            if (isApiKeyValid(apiKey)) {
+            if (apiKeyRequester.isApiKeyValid(apiKey)) {
```

Comment:
We should try to clearly separate concerns. 
Based on its name, the `ApiKeyRequester` should only request API keys, but it should not _validate_ keys. 

To be discussed if the name should be adjusted or the code should be re-organized.

Reply:
Done.

Reply:
I created two classes: ApiKeyVerifier and ApiKeyFetcher.